### PR TITLE
[Security] Verifying if the password field is null

### LINKED ISSUE #36926: Authentication provider throw `empty password` if password is null
**Symfony version(s) affected**: ~3.4 ... ~5.0

**Description**  
I made a mistake in my form with the name of `input password` 

**How to reproduce**  
```html
<form>
    <label for="username">Username</label>
    <input id="username" name="_username" type="type" />


    <label for="password">Password</label>
    <input id="password" name="password" type="password" />
</form>
```
**Possible Solution**  
https://github.com/symfony/symfony/pull/36925

### PR DESCRIPTION
**Pull request by Shine-neko**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #36926
| License       | MIT

### CODE REVIEW

**Comment by xabbuh** [comment]
Shouldn't we rather check how the credentials could end up being `null` at all instead?

**Comment by wouterj** [comment]
I think I agree with @xabbuh here. I think this should be fixed in `UsernamePasswordFormAuthenticationListener` and a `LogicException` should be thrown instead - this is not a user error (like blank password), but a developer error.

**Comment by Shine-neko** [comment]
> I think I agree with @xabbuh here. I think this should be fixed in `UsernamePasswordFormAuthenticationListener` and a `LogicException` should be thrown instead - this is not a user error (like blank password), but a developer error.

A similar thing ? https://gist.github.com/Shine-neko/76f97ee46950a480074434fa3259f797/revisions

**Comment by wouterj** [comment]
Yes, seems good. Although I think the error message can be changed, so we can give the developer guidelines on how to fix it (e.g. adding the expected password parameter named - `$this->options['password_parameter']`).

**Comment by fabpot** [comment]
Any news? It looks the gist code is more in line with what we should do.

**Comment by Shine-neko** [comment]
@fabpot  done

File: src/Symfony/Component/Security/Http/Firewall/UsernamePasswordFormAuthenticationListener.php [hunk 1]
```diff
@@ -95,6 +95,10 @@ protected function attemptAuthentication(Request $request)
             throw new BadCredentialsException('Invalid username.');
         }
 
+        if (null === $password) {
+            throw new \LogicException(sprintf('The key "%s" cannot be empty.', $this->options['password_parameter']));
```
Comment (fabpot) [inline src/Symfony/Component/Security/Http/Firewall/UsernamePasswordFormAuthenticationListener.php:99]: `The "%s" key cannot be null.`?

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Security/Http/Firewall/UsernamePasswordFormAuthenticationListener.php [hunk 2]
```diff
@@ -95,6 +95,10 @@ protected function attemptAuthentication(Request $request)
             throw new BadCredentialsException('Invalid username.');
         }
 
+        if (null === $password) {
+            throw new \LogicException(sprintf('The key "%s" cannot be null.', $this->options['password_parameter']));
```
Comment (fabpot) [inline src/Symfony/Component/Security/Http/Firewall/UsernamePasswordFormAuthenticationListener.php:99]: Maybe try to explain the problem?

`sprintf('The key password cannot be null; check that the password field name of the form matches "%s".', $this->options['password_parameter'], $this->options['password_parameter']);`

**Review by fabpot — CHANGES_REQUESTED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Shine-neko.

**Pull request merged**